sdk/storage/Azure.Storage.DataMovement.Blobs/tests/BlobSourceCheckpointDataTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+extern alias DMBlobs;
+
+using System.IO;
+using Azure.Storage.Blobs.Models;
+using DMBlobs::Azure.Storage.DataMovement.Blobs;
+using NUnit.Framework;
+
+namespace Azure.Storage.DataMovement.Blobs.Tests
+{
+    public class BlobSourceCheckpointDataTests
```

Comment:
These tests can be easily parameterized for the different `BlobType` values.

Reply:
While this is true, I'm not sure it's necessary? This is really just checking serialization/deserialization so as long as we can get a valid Blob Type in/out we can assume the others will work too? Other tests meant to test rehydration/resume will exercise the schema more as well.

Reply:
Valid. We should probably have one test in here that explicitly tests they all work, without testing the whole rehydrate mechanism.

Reply:
Just added it to `Deserialize` since that seemed easiest and covers all scenarios.